pkg/applicationprofilemanager/v1/applicationprofile_manager.go

```diff
@@ -648,11 +633,10 @@ func (am *ApplicationProfileManager) ContainerCallback(notif containercollection
 		if am.watchedContainerChannels.Has(notif.Container.Runtime.ContainerID) {
 			return
 		}
-		am.savedCapabilities.Set(k8sContainerID, mapset.NewSet[string]())
-		am.droppedEvents.Set(k8sContainerID, false)
-		am.savedEndpoints.Set(k8sContainerID, new(maps.SafeMap[string, *v1beta1.HTTPEndpoint]))
-		am.savedExecs.Set(k8sContainerID, new(maps.SafeMap[string, []string]))
-		am.savedOpens.Set(k8sContainerID, new(maps.SafeMap[string, mapset.Set[string]]))
+		am.savedCapabilities.Set(k8sContainerID, cache.NewTTL(5*am.cfg.UpdateDataPeriod, am.cfg.UpdateDataPeriod))
```

Comment:
The cache should be as long as the learning period no?

Reply:
Maybe we need a logic instead of expiring cache that when the learning period is over we delete these maps.

Reply:
We already have this logic, see the method where I added the missing endpoint maps. The idea here is to avoid keeping in memory files that we access once and we already sent to storage. The whole point of these saved maps is to avoid sending the same patches every 10 minutes to storage, at the expense of more memory

Reply:
Initially we didn't have them, but David feared we're putting too much pressure on storage